mitodl/mitxpro
# Added enrollment change status fields

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #461 

#### What's this PR do?
Adds a status field to the enrollment models that represents a change in a user's enrollment

#### How should this be manually tested?
The main purpose of this PR was simply to add the fields, but as an added fix, I made sure to return only active enrollments in various functions that fetch them, and to reactivate any inactive enrollments that were just purchased in an order. That reactivation can be tested by setting an enrollment to inactive, then running through a full checkout process with that given program/course run and checking to see if it has been reactivated.

#### Any background context you want to provide?
As mentioned above, there are two extra bits of functionality in this PR:
1. Only active enrollments are returned for various functions that fetch a user's current enrollments
1. If a user has an inactive program/course run enrollment and completes checkout with that program/course run, that enrollment will be reactivated.
